subsys/bootloader/bl_validation/bl_validation.c

```diff
@@ -267,20 +331,32 @@ static bool validate_firmware(u32_t fw_dst_address, u32_t fw_src_address,
 		return false;
 	}
 
-	if (!(((u32_t)fwinfo >= fw_src_address)
-		&& (((u32_t)fwinfo + fwinfo->total_size)
-			< (fw_src_address + fwinfo->size)))) {
+	if ((fwinfo->size > (PM_S0_SIZE))
+		|| (fwinfo->size > (PM_S1_SIZE))
+		|| (fwinfo->total_size > fwinfo->size)) {
```

Comment:
Should we subtract `CONFIG_FW_INFO_OFFSET` to ensure that the total_size does not stretch outside the fw?

Reply:
This is ensured by the region_within() call below. This is a smoke test.